### DISCUSSION TITLE
Call to a member function connection() on null on AdminPanelProvider (V3)

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#7859] [Help]
Hello,

I have the next use case: I would like to restrict the panel access to some domains, but those are stored on a database table, so I tried:

 ```php
    public function panel(Panel $panel): Panel
    {
        return $panel
            ->id('cliente')
            ->path('acceso')
            ->domains(
                Empresa::query()
                    ->pluck('subdominio')
                    ->map(fn (string $subdominio) => $subdominio . '.' . config('app.host'))
                    ->toArray()
            )
```

This results on an error:

```
Call to a member function connection() on null
```

I think this is because the panel provider is loaded before Eloquent is booted. Is there any way to change this behaviour or maybe another way to reach the database from the panel provider?

Thanks!

**Comment by danharrin** [comment]
The `domains()` method actually affects the routes that are registered for the app, so that wouldn't work. If you're trying to implement multi-tenancy, please use a package like spatie/laravel-tenancy

**Reply by underdpt** [reply]
Thanks for the answer.

It's not multi-tenancy, I only though of using `domains()` as a security enhancement :-)
